File: backend/Codebymister.Domain/Entities/Lead.cs
using Codebymister.Domain.Common;
using Codebymister.Domain.Enums;
using Codebymister.Domain.ValueObjects;

namespace Codebymister.Domain.Entities;

public class Lead : BaseEntity
{
    public string Name { get; private set; }
    public string Segment { get; private set; }
    public string City { get; private set; }
    public Url? Website { get; private set; }
    public Instagram? Instagram { get; private set; }
    public Phone? Phone { get; private set; }
    public string ProblemDescription { get; private set; }
    public LeadPriority Priority { get; private set; }
    public LeadSource Source { get; private set; }

    private Lead() { }

    public Lead(
        string name,
        string segment,
        string city,
        string problemDescription,
        LeadPriority priority,
        LeadSource source,
        string? website = null,
        string? instagram = null,
        string? phone = null)
    {
        Name = name;
        Segment = segment;
        City = city;
        Website = !string.IsNullOrWhiteSpace(website) ? Url.Create(website) : null;
        Instagram = !string.IsNullOrWhiteSpace(instagram) ? ValueObjects.Instagram.Create(instagram) : null;
        Phone = !string.IsNullOrWhiteSpace(phone) ? ValueObjects.Phone.Create(phone) : null;
        ProblemDescription = problemDescription;
        Priority = priority;
        Source = source;
    }

    public void Update(
        string name,
        string segment,
        string city,
        string problemDescription,
        LeadPriority priority,
        LeadSource source,
        string? website,
        string? instagram,
        string? phone)
    {
        Name = name;
        Segment = segment;
        City = city;
        Website = !string.IsNullOrWhiteSpace(website) ? Url.Create(website) : null;
        Instagram = !string.IsNullOrWhiteSpace(instagram) ? ValueObjects.Instagram.Create(instagram) : null;
        Phone = !string.IsNullOrWhiteSpace(phone) ? ValueObjects.Phone.Create(phone) : null;
        ProblemDescription = problemDescription;
        Priority = priority;
        Source = source;
    }
}

File: dashboard/src/types/lead.ts
export enum LeadPriority {
  Low = 1,
  Medium = 2,
  High = 3,
  VeryHigh = 4,
}

export enum LeadSource {
  GoogleSearch = 1,
  Instagram = 2,
  Facebook = 3,
  LinkedIn = 4,
  Referral = 5,
  ColdCall = 6,
  Event = 7,
  Other = 8,
}

export interface LeadDto {
  id: string;
  name: string;
  segment: string;
  city: string;
  website?: string | null;
  instagram?: string | null;
  phone?: string | null;
  problemDescription: string;
  priority: LeadPriority;
  source: LeadSource;
  createdAt: string;
  updatedAt?: string | null;
}

export interface CreateLeadRequest {
  name: string;
  segment: string;
  city: string;
  problemDescription: string;
  priority: LeadPriority;
  source: LeadSource;
  website?: string;
  instagram?: string;
  phone?: string;
}

export interface UpdateLeadRequest {
  name: string;
  segment: string;
  city: string;
  problemDescription: string;
  priority: LeadPriority;
  source: LeadSource;
  website?: string;
  instagram?: string;
  phone?: string;
}
